beacon-chain/p2p/encoder/ssz.go

```diff
@@ -110,7 +110,18 @@ func (e SszNetworkEncoder) doDecode(b []byte, to interface{}) error {
 	if v, ok := to.(fastssz.Unmarshaler); ok {
 		return v.UnmarshalSSZ(b)
 	}
-	return ssz.Unmarshal(b, to)
+	err := ssz.Unmarshal(b, to)
+	if err != nil {
+		// Check if we are unmarshalling block roots
+		// and then lop off the 4 byte offset and try
+		// unmarshalling again. This is temporary to
+		// avoid too much disruption to live nodes.
+		if _, ok := to.([][32]byte); ok {
```

Comment:
Can we add a todo which references an issue so we don't forget to remove this eventually

Reply:
Isn't this too loose? `[][32]byte` can be more than block roots right? 

Can we apply the altona flag here and only proceed with this whitelist if the altona flag is applied?

Reply:
we only have 1 rpc request type that uses `[][32]byte` which is our beacon blocks by roots request. This method is only used in rpc/gossip. And we have predefined types for all our protocols. So i think it is safe to do it here.